module-cellular/modem/ATCommon.cpp

```diff
@@ -156,3 +162,108 @@ Result Channel::checkResult(bsp::cellular::CellularResultCode cellularResult)
 
     return result;
 }
+
+std::string Channel::anonymizeIfNecessary(std::string text)
+{
+    std::string substringCPIN = "CPIN";
+    std::string substringCLCC = "CLCC";
+    std::string substringATD  = "ATD";
+    std::string substringCPBR = "CPBR";
+    std::string substringCSPN = "CSPN";
+
+    if (text.find(substringCPIN) != std::string::npos) {
+        text = anonymizeInQuotationMarks(text);
+    }
+    else if (text.find(substringCLCC) != std::string::npos) {
+        text = anonymizeInQuotationMarks(text, 2);
+    }
+    else if (text.find(substringATD) != std::string::npos) {
+        text = anonymizeNumbers(text, 2);
+    }
+    else if (text.find(substringCPBR) != std::string::npos) {
+        text = anonymizeInQuotationMarks(text, 1);
+    }
+    else if (text.find(substringCSPN) != std::string::npos) {
+        text = anonymizeInQuotationMarks(text, 2);
+    }
+
+    return text;
+}
+std::string Channel::anonymizeInQuotationMarks(std::string textToAnonymize, size_t singsToLeft)
+{
+    size_t startPosition = 0;
+    size_t endPosition   = 0;
+
+    // Find the position of the substring within the input string
+    while (startPosition < textToAnonymize.length()) {
+        startPosition = textToAnonymize.find('\"', startPosition);
+        endPosition   = textToAnonymize.find('\"', startPosition + 1);
+
+        if (startPosition != std::string::npos && endPosition != std::string::npos) {
+            // Extract the substring between the quotes
+            std::string extractedSubstring = textToAnonymize.substr(startPosition + 1, endPosition - startPosition - 1);
+
+            // Determine the length of the extracted substring
+            size_t extractedLength = extractedSubstring.length();
+
+            // Amount of signs to left can't be higher than extracted substring
+            singsToLeft = std::min(extractedLength, singsToLeft);
+
+            // Replace the substring with the appropriate number of '*' characters
+            textToAnonymize.replace(
+                startPosition + 1, extractedLength - singsToLeft, extractedLength - singsToLeft, '*');
```

Comment:
I'd also define this `'*'` as a `constexpr` - it will be easier to read and to change the anonymization character in case it's needed in the future